docs/Parameters.md
# Parameters

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**bank_fee** | **float** | Bank fees accounting at conversion | [optional] [default to 0.007]
**default_market_update_rate** | **int** | Default market&#39;s depth update rate in seconds | [optional] [default to 20]
**fiat_update_delay** | **int** | Delay in seconds between an exchange rate updates | [optional] 
**market_expiration_time** | **int** | Markets order book expiration time | [optional] 
**markets** | **list[str]** | List of market names | [optional] 
**max_tx_volume** | **float** | The max money volume that can be involved into transfer | [optional] [default to 10.0]
**observers** | **list[str]** | List of opportunity observers names | [optional] 
**refresh_rate** | **int** | Update rate in seconds of the arbiter&#39;s main loop | [optional] [default to 20]
**report_queue** | **str** | The name of the response queue | [optional] [default to 'arbitrage_watcher']

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)




docs/WatcherConfig.md
# WatcherConfig

## Properties
Name | Type | Description | Notes
------------ | ------------- | ------------- | -------------
**bank_fee** | **float** | Bank fees accounting at conversion | [optional] [default to 0.007]
**default_market_update_rate** | **int** | Default market&#39;s depth update rate in seconds | [optional] [default to 20]
**fiat_update_delay** | **int** | Delay in seconds between an exchange rate updates | [optional] 
**market_expiration_time** | **int** | Markets order book expiration time | [optional] 
**markets** | **list[str]** | List of market names | [optional] 
**max_tx_volume** | **float** | The max money volume that can be involved into transfer | [optional] [default to 10.0]
**observers** | **list[str]** | List of opportunity observers names | [optional] 
**refresh_rate** | **int** | Update rate in seconds of the arbiter&#39;s main loop | [optional] [default to 20]
**report_queue** | **str** | The name of the response queue | [optional] [default to 'arbitrage_watcher']

[[Back to Model list]](../README.md#documentation-for-models) [[Back to API list]](../README.md#documentation-for-api-endpoints) [[Back to README]](../README.md)




swagger_client/models/parameters.py
# coding: utf-8

"""
    Arbitrage Watcher RESTful API

    Provides the set of endpoints to manage arbitrage watcher service.

    OpenAPI spec version: 0.1
    
    Generated by: https://github.com/swagger-api/swagger-codegen.git
"""


from pprint import pformat
from six import iteritems
import re


class Parameters(object):
    """
    NOTE: This class is auto generated by the swagger code generator program.
    Do not edit the class manually.
    """
    def __init__(self, bank_fee=0.007, default_market_update_rate=20, fiat_update_delay=None, market_expiration_time=None, markets=None, max_tx_volume=10.0, observers=None, refresh_rate=20, report_queue='arbitrage_watcher'):
        """
        Parameters - a model defined in Swagger

        :param dict swaggerTypes: The key is attribute name
                                  and the value is attribute type.
        :param dict attributeMap: The key is attribute name
                                  and the value is json key in definition.
        """
        self.swagger_types = {
            'bank_fee': 'float',
            'default_market_update_rate': 'int',
            'fiat_update_delay': 'int',
            'market_expiration_time': 'int',
            'markets': 'list[str]',
            'max_tx_volume': 'float',
            'observers': 'list[str]',
            'refresh_rate': 'int',
            'report_queue': 'str'
        }

        self.attribute_map = {
            'bank_fee': 'bank_fee',
            'default_market_update_rate': 'default_market_update_rate',
            'fiat_update_delay': 'fiat_update_delay',
            'market_expiration_time': 'market_expiration_time',
            'markets': 'markets',
            'max_tx_volume': 'max_tx_volume',
            'observers': 'observers',
            'refresh_rate': 'refresh_rate',
            'report_queue': 'report_queue'
        }

        self._bank_fee = bank_fee
        self._default_market_update_rate = default_market_update_rate
        self._fiat_update_delay = fiat_update_delay
        self._market_expiration_time = market_expiration_time
        self._markets = markets
        self._max_tx_volume = max_tx_volume
        self._observers = observers
        self._refresh_rate = refresh_rate
        self._report_queue = report_queue

    @property
    def bank_fee(self):
        """
        Gets the bank_fee of this Parameters.
        Bank fees accounting at conversion

        :return: The bank_fee of this Parameters.
        :rtype: float
        """
        return self._bank_fee

    @bank_fee.setter
    def bank_fee(self, bank_fee):
        """
        Sets the bank_fee of this Parameters.
        Bank fees accounting at conversion

        :param bank_fee: The bank_fee of this Parameters.
        :type: float
        """

        self._bank_fee = bank_fee

    @property
    def default_market_update_rate(self):
        """
        Gets the default_market_update_rate of this Parameters.
        Default market's depth update rate in seconds

        :return: The default_market_update_rate of this Parameters.
        :rtype: int
        """
        return self._default_market_update_rate

    @default_market_update_rate.setter
    def default_market_update_rate(self, default_market_update_rate):
        """
        Sets the default_market_update_rate of this Parameters.
        Default market's depth update rate in seconds

        :param default_market_update_rate: The default_market_update_rate of this Parameters.
        :type: int
        """

        self._default_market_update_rate = default_market_update_rate

    @property
    def fiat_update_delay(self):
        """
        Gets the fiat_update_delay of this Parameters.
        Delay in seconds between an exchange rate updates

        :return: The fiat_update_delay of this Parameters.
        :rtype: int
        """
        return self._fiat_update_delay

    @fiat_update_delay.setter
    def fiat_update_delay(self, fiat_update_delay):
        """
        Sets the fiat_update_delay of this Parameters.
        Delay in seconds between an exchange rate updates

        :param fiat_update_delay: The fiat_update_delay of this Parameters.
        :type: int
        """

        self._fiat_update_delay = fiat_update_delay

    @property
    def market_expiration_time(self):
        """
        Gets the market_expiration_time of this Parameters.
        Markets order book expiration time

        :return: The market_expiration_time of this Parameters.
        :rtype: int
        """
        return self._market_expiration_time

    @market_expiration_time.setter
    def market_expiration_time(self, market_expiration_time):
        """
        Sets the market_expiration_time of this Parameters.
        Markets order book expiration time

        :param market_expiration_time: The market_expiration_time of this Parameters.
        :type: int
        """

        self._market_expiration_time = market_expiration_time

    @property
    def markets(self):
        """
        Gets the markets of this Parameters.
        List of market names

        :return: The markets of this Parameters.
        :rtype: list[str]
        """
        return self._markets

    @markets.setter
    def markets(self, markets):
        """
        Sets the markets of this Parameters.
        List of market names

        :param markets: The markets of this Parameters.
        :type: list[str]
        """

        self._markets = markets

    @property
    def max_tx_volume(self):
        """
        Gets the max_tx_volume of this Parameters.
        The max money volume that can be involved into transfer

        :return: The max_tx_volume of this Parameters.
        :rtype: float
        """
        return self._max_tx_volume

    @max_tx_volume.setter
    def max_tx_volume(self, max_tx_volume):
        """
        Sets the max_tx_volume of this Parameters.
        The max money volume that can be involved into transfer

        :param max_tx_volume: The max_tx_volume of this Parameters.
        :type: float
        """

        self._max_tx_volume = max_tx_volume

    @property
    def observers(self):
        """
        Gets the observers of this Parameters.
        List of opportunity observers names

        :return: The observers of this Parameters.
        :rtype: list[str]
        """
        return self._observers

    @observers.setter
    def observers(self, observers):
        """
        Sets the observers of this Parameters.
        List of opportunity observers names

        :param observers: The observers of this Parameters.
        :type: list[str]
        """

        self._observers = observers

    @property
    def refresh_rate(self):
        """
        Gets the refresh_rate of this Parameters.
        Update rate in seconds of the arbiter's main loop

        :return: The refresh_rate of this Parameters.
        :rtype: int
        """
        return self._refresh_rate

    @refresh_rate.setter
    def refresh_rate(self, refresh_rate):
        """
        Sets the refresh_rate of this Parameters.
        Update rate in seconds of the arbiter's main loop

        :param refresh_rate: The refresh_rate of this Parameters.
        :type: int
        """

        self._refresh_rate = refresh_rate

    @property
    def report_queue(self):
        """
        Gets the report_queue of this Parameters.
        The name of the response queue

        :return: The report_queue of this Parameters.
        :rtype: str
        """
        return self._report_queue

    @report_queue.setter
    def report_queue(self, report_queue):
        """
        Sets the report_queue of this Parameters.
        The name of the response queue

        :param report_queue: The report_queue of this Parameters.
        :type: str
        """

        self._report_queue = report_queue

    def to_dict(self):
        """
        Returns the model properties as a dict
        """
        result = {}

        for attr, _ in iteritems(self.swagger_types):
            value = getattr(self, attr)
            if isinstance(value, list):
                result[attr] = list(map(
                    lambda x: x.to_dict() if hasattr(x, "to_dict") else x,
                    value
                ))
            elif hasattr(value, "to_dict"):
                result[attr] = value.to_dict()
            elif isinstance(value, dict):
                result[attr] = dict(map(
                    lambda item: (item[0], item[1].to_dict())
                    if hasattr(item[1], "to_dict") else item,
                    value.items()
                ))
            else:
                result[attr] = value

        return result

    def to_str(self):
        """
        Returns the string representation of the model
        """
        return pformat(self.to_dict())

    def __repr__(self):
        """
        For `print` and `pprint`
        """
        return self.to_str()

    def __eq__(self, other):
        """
        Returns true if both objects are equal
        """
        if not isinstance(other, Parameters):
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        """
        Returns true if both objects are not equal
        """
        return not self == other
